test/extended/machine_config/machine_config_node.go

```diff
@@ -0,0 +1,388 @@
+	o.Expect(clientErr).NotTo(o.HaveOccurred(), "Error creating client set for test.")
+
+	// In SNO, master pool will degrade
+	poolName := worker
+	mcName := "91-worker-testfile-invalid"
+	if isSno {
+		poolName = master
+		mcName = "91-master-testfile-invalid"
+	}
+
+	// Cleanup MC and fix node degradation on failure or test completion
+	defer func() {
+		// Delete the applied MC
+		deleteMCErr := oc.Run("delete").Args("machineconfig", mcName).Execute()
+		o.Expect(deleteMCErr).NotTo(o.HaveOccurred(), fmt.Sprintf("Could not delete MachineConfig '%v'.", mcName))
+
+		// Recover the degraded MCP
+		recoverErr := RecoverFromDegraded(oc, poolName)
+		o.Expect(recoverErr).NotTo(o.HaveOccurred(), fmt.Sprintf("Could not recover MCP '%v' from degraded state.", poolName))
+	}()
+
+	// Apply invalid MC
+	mcErr := oc.Run("apply").Args("-f", fixture).Execute()
+	o.Expect(mcErr).NotTo(o.HaveOccurred(), "Could not apply MachineConfig.")
+
+	// Wait for MCP to be in a degraded state with one degraded machine
+	degradedErr := WaitForMCPConditionStatus(oc, poolName, "Degraded", corev1.ConditionTrue, 8*time.Minute, 3*time.Second)
+	o.Expect(degradedErr).NotTo(o.HaveOccurred(), fmt.Sprintf("Error waiting for '%v' MCP to be in a degraded state.", poolName))
+	mcp, err := clientSet.MachineconfigurationV1().MachineConfigPools().Get(context.TODO(), poolName, metav1.GetOptions{})
+	o.Expect(err).NotTo(o.HaveOccurred(), fmt.Sprintf("Error getting '%v' MCP.", poolName))
+	o.Expect(mcp.Status.DegradedMachineCount).To(o.BeNumerically("==", 1), fmt.Sprintf("Degraded machine count is not 1. It is %v.", mcp.Status.DegradedMachineCount))
+
+	// Get degraded node
+	degradedNode, degradedNodeErr := GetDegradedNode(oc, poolName)
+	o.Expect(degradedNodeErr).NotTo(o.HaveOccurred(), "Could not get degraded node.")
+
+	// Validate MCN of degraded node
+	degradedNodeMCN, degradedErr := clientSet.MachineconfigurationV1alpha1().MachineConfigNodes().Get(context.TODO(), degradedNode.Name, metav1.GetOptions{})
+	o.Expect(degradedErr).NotTo(o.HaveOccurred(), fmt.Sprintf("Error getting MCN of degraded node '%v'.", degradedNode.Name))
+	framework.Logf("Validating that `AppliedFilesAndOS` and `UpdateExecuted` conditions in '%v' MCN have a status of 'Unknown'.", degradedNodeMCN.Name)
+	o.Expect(CheckMCNConditionStatus(degradedNodeMCN, mcfgv1alpha1.MachineConfigNodeUpdateFilesAndOS, metav1.ConditionUnknown)).Should(o.BeTrue(), "Condition 'AppliedFilesAndOS' does not have the expected status of 'Unknown'.")
+	o.Expect(CheckMCNConditionStatus(degradedNodeMCN, mcfgv1alpha1.MachineConfigNodeUpdateExecuted, metav1.ConditionUnknown)).Should(o.BeTrue(), "Condition 'UpdateExecuted' does not have the expected status of 'Unknown'.")
+}
+
+// `ValidateMCNProperties` checks that MCNs with correct properties are created on node creation
+// and deleted on node deletion
+func ValidateMCNOnNodeCreationAndDeletion(oc *exutil.CLI) {
+	cleanupCompleted := false
+	newNode := corev1.Node{}
+	newMachineName := ""
```

Comment:
```suggestion
	var cleanupCompleted bool
```
Default value for bools are false, but this works too if you want to be explicit about it. 
For `newNode` and `newMachineName`, you could do the standard initialization on assignment. You aren't using them in defer blocks prior to that.

Reply:
I think I'd prefer to explicitly set `cleanupCompleted` false, but I cleaned up the other unnecessary variables. Thanks!